cdc/entry/mounter.go

```diff
@@ -299,38 +312,77 @@ func IsLegacyFormatJob(rawKV *model.RawKVEntry) bool {
 }
 
 // ParseDDLJob parses the job from the raw KV entry. id is the column id of `job_meta`.
-func ParseDDLJob(tblInfo *model.TableInfo, rawKV *model.RawKVEntry, id int64) (*timodel.Job, error) {
+func ParseDDLJob(rawKV *model.RawKVEntry, ddlTableInfo *DDLTableInfo) (*timodel.Job, error) {
 	var v []byte
+	var datum types.Datum
+
+	// for test case only
 	if bytes.HasPrefix(rawKV.Key, metaPrefix) {
-		// old queue base job.
 		v = rawKV.Value
-	} else {
-		// DDL job comes from `tidb_ddl_job` table after we support concurrent DDL. We should decode the job from the column.
-		recordID, err := tablecodec.DecodeRowKey(rawKV.Key)
-		if err != nil {
-			return nil, errors.Trace(err)
+		job, err := parseJob(v, rawKV.StartTs, rawKV.CRTs, false)
+		if err != nil || job == nil {
+			job, err = parseJob(v, rawKV.StartTs, rawKV.CRTs, true)
 		}
-		row, err := decodeRow(rawKV.Value, recordID, tblInfo, time.UTC)
+		return job, err
+	}
+
+	// DDL job comes from `tidb_ddl_job` table after we support concurrent DDL. We should decode the job from the column.
+	recordID, err := tablecodec.DecodeRowKey(rawKV.Key)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	// first parse it with tidb_ddl_job, if failed, parse it with tidb_ddl_history
```

Comment:
Maybe you can decode the tableID from the key?

Reply:
Good Suggestion! Fixed now.